libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -318,12 +320,14 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_ReadSyncAssisted )
     /* restore slave select */
     if( ulAssistedTestIotSpiSlave != ultestIotSpiSlave )
     {
-        lRetVal = iot_spi_select_slave( ultestIotSpiInstance, ulAssistedTestIotSpiSlave );
-        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        #ifdef IOT_TEST_COMMON_IO_SPI_SLAVE_SELECT_SUPPORTED
+            lRetVal = iot_spi_select_slave( ulAssistedTestIotSpiInstance, ulAssistedTestIotSpiSlave );
+            TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        #endif
     }
 
     /* Create a string with read bytes and print it to console. */
-    for( int i = 0, j = 0, len = sizeof( ucRxBuf ); i < len; i++ )
+    for( i = 0, j = 0, len = sizeof( ucRxBuf ); i < len; i++ )
```

Comment:
Could this loop better ensure that a future change doesn't cause us to write past the end of cMsg?

Reply:
ok